### PR TITLE
OS-8484 Headnode NTP config should be able to support RAN networks

### PR DESCRIPTION
**Pull request by bahamat**
Companion: TritonDataCenter/illumos-joyent#453

### CODE REVIEW

File: src/smartdc/lib/ntp_config.js [hunk 1]
```diff
@@ -199,9 +205,16 @@ generate_file(servers, allowed_subnets, trusted, orphan)
 				process.exit(1);
 			}
 
-			out.push(mod_util.format('%s %s mask %s',
-			    fam === 4 ? 'restrict' : 'restrict -6', as[0],
-			    mask));
+			/*
+			 * We should deduplicate these networks since the -a
```
Comment (danmcd) [inline src/smartdc/lib/ntp_config.js:209]: Is this a "We don't do it now, but we should," sort of comment?  If so, please mark with TODO:

```
/*
 * TODO: We should deduplicate...
```

If this code below actually does the dedup (doesn't *seem* to but I might be missing something), lose the "should" as we're actually deduplicating:

```
/*
 * We deduplicate these networks...
```

File: src/smartdc/lib/ntp_config.js [hunk 2]
```diff
@@ -1,6 +1,11 @@
 #!/usr/node/bin/node
 /* vim: syn=javascript ts=8 sts=8 sw=8 noet: */
 
+/*
```
Comment (danmcd) [inline src/smartdc/lib/ntp_config.js:4]: Also, is this under some sort of license (CDDL1.0 or MPL2.0)?

File: src/smartdc/lib/ntp_config.js (hunk 1 above)
Comment (bahamat) [reply src/smartdc/lib/ntp_config.js]: Clarified this comment.

File: src/smartdc/lib/ntp_config.js (hunk 2 above)
Comment (bahamat) [reply src/smartdc/lib/ntp_config.js:4]: Added CDDL license header.

Comment (bahamat) [reply src/smartdc/lib/ntp_config.js:4]: Also, CDDL vs MPL here:

The reason for CDDL, with smartos-live, the convention has been for files that natively live in the repo (i.e., are not copied in from elsewhere), things tend to be CDDL. This file predates Joyent's significant use of MPL, so it seems the original intention would have been CDDL.

File: src/smartdc/lib/ntp_config.js [hunk 3]
```diff
@@ -1,6 +1,32 @@
...
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file.
+ *
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ *
+ * Copyright 2019 Joyent, Inc.
```
Comment (danmcd) [inline src/smartdc/lib/ntp_config.js:26]: Last nit -> Copyright block is its own block after license, typically. Also the CDDL 1.0 text is shorter, see $ILLUMOS/$SRC/prototypes/prototype.c for the example.

**Review by danmcd — COMMENTED** [review]
One last nit.  WOuld approve otherwise, thanks for the clarification.

File: src/smartdc/lib/ntp_config.js (hunk 3 above)
Comment (bahamat) [reply src/smartdc/lib/ntp_config.js:26]: Oh, ok. I just `cat`'d a file at random and did it the same way.

Comment (danmcd) [reply src/smartdc/lib/ntp_config.js:26]: Upon further review... other .js files follow this convention (from way back I'd guess).

**Review by danmcd — APPROVED** [review]

File: src/smartdc/lib/ntp_config.js (hunk 3 above)
Comment (bahamat) [reply src/smartdc/lib/ntp_config.js:26]: After looking at the rest of the repo, *most* files use the "expanded" CDDL header that I used here, and there are only a handful with separate copyright block. After discussing on Slack, we've decided that it's ok to leave this as-is.

**Pull request merged**